### ISSUE TITLE
[Bug] VLC backup goes on indefinitely

### ISSUE THREAD

**Issue by opusforlife2** [machiav3lli/oandbackupx#218]
**Description**
When a Data or 'APK and Data' backup of VLC is initiated, it goes on indefinitely. There is no error message. You have to swipe away the app to stop the backup. Since it is a failed backup, it doesn't get saved and doesn't show up the next time you open the app. A simple APK backup completes successfully.

**Steps To Reproduce**
1. Tap on Data or Both when backing up VLC.

**Expected behavior**
A successful backup.

**System Information(please complete the following information):**
 - Device: Xiaomi Mi 5
 - Android Version: 9
 - ROM: LOS 16
 - App's Version: 4.0.0

**Comment by apbucaneg** [comment]
Also getting this.

Xiaomi redmi note 9 pro
Android 10
Resurrection Remix
V 4.0.0

**Comment by fantostisch** [comment]
It seems to get stuck at https://github.com/machiav3lli/oandbackupx/blob/a2c2bab4e922963aec22f9d5fe88654d0d406665/app/src/main/java/com/machiav3lli/backup/handler/ShellHandler.kt#L319 Probably the same issue as https://github.com/topjohnwu/libsu/issues/62

**Comment by fantostisch** [comment]
It seems to get stuck when there is a file in a folder which contains a `'`.
```
2020-11-16 13:34:56.013 10825-11570/com.machiav3lli.backup.debug D/OAndBackupX.TarUtils: Adding app_vlc/.cache/art/artistalbum/assf'ss/test.txt to archive (filesize: 0)
2020-11-16 13:34:56.014 10825-11570/com.machiav3lli.backup.debug D/SHELL_IN: __F_='/data/user/0/org.videolan.vlc/app_vlc/.cache/art/artistalbum/assf'ss/test.txt'
2020-11-16 13:34:56.014 10825-11570/com.machiav3lli.backup.debug D/SHELL_IN: [ -b "$__F_" ]
```
When removing the `'` from the folder name the backup is successful.

**Comment by KurtDoedel** [comment]
@fantostisch Simply replacing `"'"` with `"\\'"` results in the single quotes the file name argument to `dd` is enclosed in not being closed (try `echo 'hello \' there'` in a Bash shell yourself). Would you mind trying `.replace("'", "'\"'\"'")` or `.replace("'", "'\\''")` instead? Read [this StackOverflow article on escaping single quotes inside single quotes](https://stackoverflow.com/questions/1250079/how-to-escape-single-quotes-within-single-quoted-strings) for further information.

**Comment by tsiflimagas** [comment]
The issue is probably fixed in version 5.0.1.

**Comment by opusforlife2** [comment]
Whoops. Forgot to close this. See https://github.com/machiav3lli/oandbackupx/issues/205#issuecomment-821772886.
